src/main/kotlin/org/jetbrains/research/ml/coding/assistant/tree/TreeConverter.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2020.  Anastasiia Birillo
+ */
+
+package org.jetbrains.research.ml.coding.assistant.tree
+
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.TreeContext
+import com.github.gumtreediff.tree.TreeUtils
+import com.intellij.psi.PsiElement
+import util.PsiUtil.label
+import util.PsiUtil.preOrderNumbering
+import java.util.*
+
+object TreeConverter {
+
+    /*
+    * Create a GumTree tree for PSI.  We use BFS to traversal the PSI. To indicate the level is ended we add null
+    * into psiNodes. The algorithm works as follow:
+    * Let we have a tree:
+    *      [1]
+    *   /   |   \
+    * [2]  [3]  [4]
+    *  | \
+    * [5] [6]
+    * ...
+    * Each vertex from the list of vertex [2, 3, 4] we should link with the vertex 1.
```

Comment:
I'm not sure, but maybe this 'We should link each vertex ...' or this 'Each vertex .. should be linked' sounds better because of words ordering (in most cases it should be subject + verb + object)

Reply:
(don't need to fix since the implementation will change)